go/consensus/tendermint/apps/staking/state/state.go

```diff
@@ -624,55 +624,55 @@ func slashPool(dst *quantity.Quantity, p *staking.SharePool, amount, total *quan
 
 // SlashEscrow slashes the escrow balance and the escrow-but-undergoing-debonding
 // balance of the account, transferring it to the global common pool, returning
-// true iff the amount actually slashed is > 0.
+// the amount actually slashed.
 //
 // WARNING: This is an internal routine to be used to implement staking policy,
 // and MUST NOT be exposed outside of backend implementations.
 func (s *MutableState) SlashEscrow(
 	ctx *abciAPI.Context,
 	fromAddr staking.Address,
 	amount *quantity.Quantity,
-) (bool, error) {
+) (*quantity.Quantity, error) {
+	var slashed quantity.Quantity
+
 	commonPool, err := s.CommonPool(ctx)
 	if err != nil {
-		return false, fmt.Errorf("tendermint/staking: failed to query common pool for slash: %w", err)
+		return nil, fmt.Errorf("tendermint/staking: failed to query common pool for slash: %w", err)
 	}
 
 	from, err := s.Account(ctx, fromAddr)
 	if err != nil {
-		return false, fmt.Errorf("tendermint/staking: failed to query account %s: %w", fromAddr, err)
+		return nil, fmt.Errorf("tendermint/staking: failed to query account %s: %w", fromAddr, err)
 	}
 
 	// Compute the amount we need to slash each pool. The amount is split
 	// between the pools based on relative total balance.
 	total := from.Escrow.Active.Balance.Clone()
 	if err = total.Add(&from.Escrow.Debonding.Balance); err != nil {
-		return false, fmt.Errorf("tendermint/staking: compute total balance: %w", err)
+		return nil, fmt.Errorf("tendermint/staking: account total balance: %w", err)
+	}
+	// Nothing to slash.
+	if total.IsZero() {
```

Comment:
Note that before this check was bellow (after `slashPool`) calls.

However in case nothing could've been slashed (node has no stake) the `slashPool` call failed already (since it divides by `total` which would be zero), making the bellow check ineffective.